diff --git a/lib/window/VMGlobalPropertyScript.js b/lib/window/VMGlobalPropertyScript.js
index d44a1fb938711f0146216900fd2d5ce861b426eb..c17618149c0dfd198ac68b307c5ae402e23f18e5 100644
--- a/lib/window/VMGlobalPropertyScript.js
+++ b/lib/window/VMGlobalPropertyScript.js
@@ -47,7 +47,6 @@ this.isFinite = globalThis.isFinite;
 this.isNaN = globalThis.isNaN;
 this.parseFloat = globalThis.parseFloat;
 this.parseInt = globalThis.parseInt;
-this.process = null;
 this.root = globalThis.root;
 this.undefined = globalThis.undefined;
 this.unescape = globalThis.unescape;
